scripts/force_layer_download.py

```diff
@@ -0,0 +1,235 @@
+import argparse
+import asyncio
+import json
+import logging
+import sys
+from typing import Any, List, Tuple
+
+import aiohttp
+
+
+class ClientException(Exception):
+    pass
+
+
+class Client:
+    def __init__(self, pageserver_api_endpoint: str):
+        self.sess = aiohttp.ClientSession(pageserver_api_endpoint)
+
+    async def close(self):
+        await self.sess.close()
+
+    async def get_tenant_ids(self):
+        resp = await self.sess.get("/v1/tenant")
+        body = await resp.json()
+        if not resp.ok:
+            raise ClientException(f"{resp}")
+        if not isinstance(body, list):
+            raise ClientException("expecting list")
+        return [t["id"] for t in body]
+
+    async def get_timeline_ids(self, tenant_id):
+        resp = await self.sess.get(f"/v1/tenant/{tenant_id}/timeline")
+        body = await resp.json()
+        if not resp.ok:
+            raise ClientException(f"{resp}")
+        if not isinstance(body, list):
+            raise ClientException("expecting list")
+        return [t["timeline_id"] for t in body]
+
+    async def timeline_spawn_download_remote_layers(self, tenant_id, timeline_id, ongoing_ok=False):
+
+        resp = await self.sess.post(
+            f"/v1/tenant/{tenant_id}/timeline/{timeline_id}/download_remote_layers",
+        )
+        body = await resp.json()
+        if resp.status == 409:
+            if not ongoing_ok:
+                raise ClientException("download already ongoing")
+            pass  # response body has same shape for ongoing and newly created
+        elif not resp.ok:
+            raise ClientException(f"{resp}")
+
+        if not isinstance(body, dict):
+            raise ClientException("expecting dict")
+
+        return body
+
+    async def timeline_poll_download_remote_layers_status(
+        self,
+        tenant_id,
+        timeline_id,
+    ):
+        resp = await self.sess.get(
+            f"/v1/tenant/{tenant_id}/timeline/{timeline_id}/download_remote_layers",
+        )
+        body = await resp.json()
+
+        if not resp.ok:
+            raise ClientException(f"{resp}")
+
+        return body
+
+    async def timeline_download_remote_layers(
+        self,
+        tenant_id,
+        timeline_id,
+    ):
+        spawn_res = await self.timeline_spawn_download_remote_layers(
+            tenant_id, timeline_id, ongoing_ok=False
+        )
+        while True:
+            st = await self.timeline_poll_download_remote_layers_status(tenant_id, timeline_id)
+            logging.info("{tenant_id}:{timeline_id} state is: {st}")
+
+            if spawn_res["task_id"] != st["task_id"]:
+                raise ClientException("download task ids changed while polling")
+
+            if st["state"] != "Completed":
+                await asyncio.sleep(10)
+                continue
+
+            return st
+
+
+class Completed(dict[str, Any]):
```

Comment:
it's not usual to inherit from `dict`. It's better to do sth like this.

```python
@dataclass
class Completed:
    result: Dit[str, Any]
```

and use 

```python
x = Completed(task_result)
print(x.result["state"])
```